data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -527,23 +527,30 @@ private String createIndexes(PersistentEntity entity, String tableName) {
         Stream.of(indexes)
                 .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
                 .flatMap(Collection::stream)
-                .forEach(index -> addIndex(indexBuilder, new IndexConfiguration(index, tableName)));
+                .forEach(index -> indexStatements.add(addIndex(entity, new IndexConfiguration(index, tableName, entity.getPersistedName()))));
 
-        return indexBuilder.toString();
+        return indexStatements;
 
     }
 
-    private void addIndex(StringBuilder indexBuilder, IndexConfiguration config) {
-        indexBuilder.append("CREATE ")
-                .append(config.index.booleanValue("unique")
+    private String addIndex(PersistentEntity entity, IndexConfiguration config) {
+        // Create index name without escaped table name and then escape if needed
+        StringBuilder sbIndexName = new StringBuilder();
```

Comment:
Current code was producing some weird errors when creating indexes so this fixes it. For example index would be named 
```
idx_`shipment`_on_delivery(delivery)
```